src/execution.ts

```diff
@@ -940,14 +947,22 @@ function handleArrayValue(
  * Implements a generic map operation for any iterable.
  *
  * If the iterable is not valid, null is returned.
+ * @param reuseArray true when the input array should be reused
  * @param {Iterable<any> | string} iterable possible iterable
  * @param {(a: any) => any} cb callback that receives the item being iterated
  * @returns {any[]} a new array with the result of the callback
  */
 function safeMap(
+  reuseArray: boolean,
   iterable: Iterable<any> | string,
   cb: (a: any, idx: number) => any
 ): any[] {
+  if (reuseArray && Array.isArray(iterable)) {
```

Comment:
IMO, it feels like this optimization probably hurts more than giving improvements. 
https://v8.dev/blog/elements-kinds#avoid-elements-kind-transitions

Are you seeing benefits reusing the array?

Reply:
Yes, the element kind will be same on the transition so there will not be any degradation.

I see double throughput on the nested arrays benchmark.

Reply:
cool 👍